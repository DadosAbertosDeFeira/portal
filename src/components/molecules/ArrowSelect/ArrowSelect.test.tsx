/* eslint-disable testing-library/no-wait-for-empty-callback */
import { faker } from "@faker-js/faker";
import { render, waitFor } from "@testing-library/react";
import dynamic from "next/dynamic";

import { type ArrowSelectProps } from ".";

const ArrowSelect = dynamic<ArrowSelectProps<any>>(
  () => import("molecules/ArrowSelect").then((module) => module.ArrowSelect),
  { ssr: false }
);

describe("ArrowSelect", () => {
  // const itemsStringMock = Array.from(Array(5).keys()).map(() =>
  //   faker.person.fullName()
  // );

  const itemsObjectMock = Array.from(Array(5).keys()).map(() => ({
    name: faker.person.fullName(),
  }));

  function makeSut<T>({ items = [], ...props }: Partial<ArrowSelectProps<T>>) {
    return render(<ArrowSelect items={items} {...props} />);
  }

  it("renders correctly", async () => {
    const { asFragment } = makeSut({});

    await waitFor(async () => {});

    expect(asFragment).toMatchSnapshot();
  });

  it("should renders correctly with objects", async () => {
    const { asFragment } = makeSut({
      items: itemsObjectMock,
      itemToString: (v) => v?.name ?? "",
    });

    await waitFor(async () => {});

    expect(asFragment).toMatchSnapshot();
  });
});
