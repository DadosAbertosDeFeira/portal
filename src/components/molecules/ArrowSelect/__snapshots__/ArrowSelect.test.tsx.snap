// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArrowSelect renders correctly 1`] = `[Function]`;

exports[`ArrowSelect should renders correctly with objects 1`] = `[Function]`;
